File: ATM_Java_Sockets_MySQL/ServerProtocol.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class ServerProtocol {
    /*
    ERR1: user not found
    ERR2: wrong value
    ERR3: balance less than withdraw price
    ERR4: wrong action
    ERR5: DB error
    */

    public String processRequest(String inmsg){
        String outmsg;
        String[] parts = inmsg.split(" ");
        String customer_id = parts[0];
        String action = parts[1];
        String pricetext;
        double price = -1.0;
        if(action.equals("deposit") || action.equals("withdraw")){
            pricetext = parts[2];
            price = Float.parseFloat(pricetext);
            if(price < 0.0) return "ERR2";
        }

        //connect to database
        Connection conn;
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/atm", "root", "rootroot");
        } catch (SQLException e) {
            System.out.println(e.getMessage());
            System.out.println("DB Connection error");
            outmsg = "ERR5";
            return outmsg;
        } catch (ClassNotFoundException e){
            System.out.println(e.getMessage());
            outmsg = "ERR5";
            return outmsg;
        }


        switch(action){
			case "balance":
				outmsg = Database.balance(conn, customer_id);
				break;
			case "deposit":
				outmsg = Database.deposit(conn, customer_id, price);
				break;
			case "withdraw":
				outmsg = Database.withdraw(conn, customer_id, price);
				break;
			default:
				outmsg = "ERR4";
				break;
		}
        return outmsg;
    }



}

File: ATM_Python_Sockets_MySQL/ServerProtocol.py
import mysql.connector
import Database

# ERR1: user not found
# ERR2: wrong value
# ERR3: balance less than withdraw price
# ERR4: wrong action
# ERR5: DB error

def processRequest(inmsg):
	parts = str(inmsg).split()
	customer_id = parts[0]
	action = parts[1]
	price = -1.0
	if action == "deposit" or action == "withdraw":
		price = float(parts[2])
		if price < 0.0:
			return "ERR2"

	#connect to database
	try:
		connection = mysql.connector.connect(host="localhost", user="root", passwd="rootroot")
	except:
		print("DB Connection error")
		return "ERR5"
	
	if action == "balance":
		return Database.balance(connection, customer_id)
	elif action == "deposit":
		return Database.deposit(connection, customer_id, price)
	elif action == "withdraw":
		return Database.withdraw(connection, customer_id, price)
	else:
		return "ERR4"
	
	


	